### ISSUE TITLE
Spurious RS2003 error on latest 7.0 SDK

### ISSUE THREAD

**Issue by roji** [dotnet/roslyn-analyzers#5867]
When building EF Core with SDK 7.0.100-preview.2.22108.9, I get:

> CSC : error RS2003: Rule 'EF1001' was shipped in analyzer release '3.0.0', but is no longer a supported diagnostic for any analyzer. Add an entry for this rule in a 'Removed Rules' table to unshipped file. [/home/roji/projects/efcore/src/EFCore.Analyzers/EFCore.Analyzers.csproj]

Contents of AnalyzerReleases.Shipped.md:

```
## Release 2.1.0

### New Rules
Rule ID | Category | Severity | Notes
--------|----------|----------|-------
EF1000  | Usage    | Warning  | RawSqlStringInjectionDiagnosticAnalyzer, [Documentation](https://docs.microsoft.com/ef/core/querying/raw-sql)

## Release 3.0.0

### New Rules
Rule ID | Category | Severity | Notes
--------|----------|----------|-------
EF1001  | Usage    | Warning  | InternalUsageDiagnosticAnalyzer

### Removed Rules
Rule ID | Category | Severity | Notes
--------|----------|----------|--------------------
EF1000  | Security | Disabled | RawSqlStringInjectionDiagnosticAnalyzer, [Documentation](https://docs.microsoft.com/ef/core/querying/raw-sql)
```

No changes were done since EF Core 3.0, and the EF1001 diagnostic is very much alive.

To repro, clone https://github.com/dotnet/efcore, go into src/EFCore.Analyzers and try to build.

**Comment by Youssef1313** [comment]
@roji Are you using target-typed new? If yes, then this is a known issue. For now you'll have to revert target-typed new changes.

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/5828

**Comment by roji** [comment]
Thanks @Youssef1313 - yeah, we're using target-typed new exactly as described in #5828. Closing as dup.
